### ISSUE TITLE
TRAMP support

### ISSUE THREAD

**Issue by PythonNut** [raxod502/apheleia#33]
As noted in the README, TRAMP is currently unsupported. I am considering adding support since I use TRAMP extensively. I'm interested in your thoughts on this before I potentially embark on creating a PR. :slightly_smiling_face:

**Comment by raxod502** [comment]
Yep, sounds like a great enhancement. I don't have any specific thoughts and I think you know more than I do about the implications, performance and otherwise, with running operations on remote files.

**Comment by mohkale** [comment]
I'm in a situation where I'd like this as well. I recently worked on remote-file support for my `flymake-rest` project. There I added an option where users could configure how remote files are handled. The possible values were cancel, remote and local. If remote the checker is run over tramp (but that'll be blocking). If local then we run the process on your current machine. And cancel just aborts the check immediately if the file is remote.

Would something like that work for apheleia as well? There's also considerations for whether this should be on a formatter by formatter basis or just globally. Furthermore if the formatter depends on a physical filepath we'll probably have to fail it if it has to be run locally.
